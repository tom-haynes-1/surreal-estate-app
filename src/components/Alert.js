import React from 'react';
import PropTypes from 'prop-types';
import '../styles/alert.css';

const Alert = ({ message, success }) => {
  if (!message) return null;

  return (
    <div className={`alert ${success ? 'success' : 'error'}`}>
      {message}
    </div>
  );
};

export default Alert;

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  success: false,
};
