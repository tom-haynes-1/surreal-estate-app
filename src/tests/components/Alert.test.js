import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../../components/Alert';

describe('Alert component', () => {
  const validProps = {
    initial: {
      message: '',
      isSuccess: true,
    },
    error: {
      message: 'Error',
      isSuccess: false,
    },
    success: {
      message: 'Success',
      isSuccess: true,
    },
  };

  test('renders initial state correctly', () => {
    const { asFragment } = render(
      <Alert
        message={validProps.initial.message}
        success={validProps.initial.isSuccess}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correct error message', () => {
    const { asFragment, getByText } = render(
      <Alert
        message={validProps.error.message}
        success={validProps.error.isSuccess}
      />,
    );

    const renderedMessage = getByText('Error');

    expect(asFragment()).toMatchSnapshot();
    expect(renderedMessage).toBeInTheDocument();
    expect(renderedMessage).toHaveClass('alert error');
  });

  test('renders correct success message', () => {
    const { asFragment, getByText } = render(
      <Alert
        message={validProps.success.message}
        success={validProps.success.isSuccess}
      />,
    );

    const renderedMessage = getByText('Success');

    expect(asFragment()).toMatchSnapshot();
    expect(renderedMessage).toBeInTheDocument();
    expect(renderedMessage).toHaveClass('alert success');
  });
});
