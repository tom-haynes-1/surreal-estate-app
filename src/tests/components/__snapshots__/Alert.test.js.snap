// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders correct error message 1`] = `
<DocumentFragment>
  <div
    class="alert error"
  >
    Error
  </div>
</DocumentFragment>
`;

exports[`Alert component renders correct success message 1`] = `
<DocumentFragment>
  <div
    class="alert success"
  >
    Success
  </div>
</DocumentFragment>
`;

exports[`Alert component renders initial state correctly 1`] = `<DocumentFragment />`;
